src/Sylius/Bundle/CoreBundle/Doctrine/ORM/PaymentRepository.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Doctrine\ORM;
+
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+
+class PaymentRepository extends EntityRepository
+{
+    /**
+     * Create filter paginator.
+     *
+     * @param array $criteria
+     * @param array $sorting
+     *
+     * @return PagerfantaInterface
+     */
+    public function createFilterPaginator($criteria = array(), $sorting = array())
+    {
+        $this->_em->getFilters()->disable('softdeleteable');
+
+        $queryBuilder = $this->getCollectionQueryBuilder();
+
+        $queryBuilder
+            ->leftJoin($this->getAlias().'.order', 'paymentOrder')
+            ->leftJoin('paymentOrder.billingAddress', 'address')
+            ->addSelect('paymentOrder')
+            ->addSelect('address')
+        ;
+
+        if (!empty($criteria['number'])) {
+            $queryBuilder
+                ->andWhere('paymentOrder.number = :number')
+                ->setParameter('number', $criteria['number'])
+            ;
+        }
+        if (!empty($criteria['channel'])) {
+            $queryBuilder
+                ->andWhere('paymentOrder.channel = :channel')
+                ->setParameter('channel', $criteria['channel'])
+            ;
+        }
+        if (!empty($criteria['billingAddress'])) {
+            $queryBuilder
+                ->andWhere('address.lastName LIKE :billingAddress')
+                ->setParameter('billingAddress', '%'.$criteria['billingAddress'].'%')
+            ;
+        }
+        if (!empty($criteria['createdAtFrom'])) {
+            $queryBuilder
+                ->andWhere($queryBuilder->expr()->gte($this->getAlias().'.createdAt', ':createdAtFrom'))
```

Comment:
You should use `getPropertyName('createdAt')` instead of `$this->getAlias().'.createdAt'`